pkg/internal/ibmblockcsi/default_setter.go

```diff
@@ -18,31 +18,107 @@ package ibmblockcsi
 
 import (
 	"github.com/IBM/ibm-block-csi-operator/pkg/config"
+	corev1 "k8s.io/api/core/v1"
+	"path"
 )
 
 // SetDefaults set defaults if omitted in spec, returns true means CR should be updated on cluster.
 // Replace it with kubernetes native default setter when it is available.
 // https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#defaulting
 func (c *IBMBlockCSI) SetDefaults() bool {
```

Comment:
throughout the file there is `set` functions that mainly check if set is necessary and only if it is (not always) they preform the set operation.
I think we need to split the `set`s to  "check if set needed" and the set itself.

@zingero what do you think?

Reply:
Can you provide an example for this change?

Reply:
```
if is_set_needed():
    SetDefaults()
```

Reply:
@ArbelNathan,
I'm not sure how it's related to this commit. I suggest talking about it offline.

Reply:
plz open a ticket to discuss that

Reply:
I opened CSI-2826